apps/meteor/client/views/room/Header/Omnichannel/QuickActions/hooks/useQuickActions.tsx

```diff
@@ -68,15 +69,43 @@ export const useQuickActions = (
 		}
 	});
 
+	const getVisitorsVerificationStatus = useCallback(async (): Promise<boolean> => {
+		if (!visitorRoomId) {
+			return false;
+		}
+
+		const {
+			visitor: { visitorEmails },
+		} = await getVisitorInfo({ visitorId: visitorRoomId });
+
+		if (visitorEmails?.length && visitorEmails[0]?.verified === 'Verified') {
+			setVisitorsVerificationStatus(true);
+			return true;
+		}
+		setVisitorsVerificationStatus(false);
+		return false;
+	}, [getVisitorInfo, visitorRoomId]);
+
 	useEffect(() => {
 		if (onHoldModalActive && roomLastMessage?.token) {
 			setModal(null);
 		}
-	}, [roomLastMessage, onHoldModalActive, setModal]);
+		getVisitorsVerificationStatus();
```

Comment:
I won't recommend using useEffect for checking verification status as it can lead to unknown states due to re-rendering. Instead we can make use of Tanstack's useQuery() to fetch visitor info from visitors.info. 
PS: Considering my other suggestion in regards to storing the verification status on Room level, you might have to fetch this info from an another endpoint